### PR TITLE
feat: add "Process Emails Now" button and auto-trigger email processi…

### LINKED ISSUE #2920: Feature Request: Process Emails button
Current mailer behavior runs `esp/dbmail_cron.py` for new emails every ~15 minutes and then sends them. This behavior is fine for mass mailing however is a bit of a pain when sending test emails as the user may have to wait up to 15 minutes for their test email to arrive. In the case that changes need to make this can turn into a very lengthy process. 

Request would be to add a button that processes emails immediately.

### PR DESCRIPTION
**Pull request by abhinavjha0239**
## Description

The email system uses a cron job (`dbmail_cron.py`) that runs every ~15 minutes to process and send queued emails. When admins send test emails via the Communications Panel, they have to wait up to 15 minutes for delivery — making iterative testing painfully slow.

This PR adds **automatic email processing on commpanel submit** (as the maintainer suggested) plus **manual "Process Emails Now" buttons** on both the email monitoring dashboard and the commpanel finished page as fallbacks.

### How it works

When an admin submits an email through the Communications Panel, `dbmail_cron.py` is automatically spawned as a **detached subprocess**. The cron script's existing `fcntl` file locking prevents concurrent execution — if the cron job (or another button press) is already running, the subprocess exits harmlessly with no side effects.

**Why both auto-trigger AND manual buttons:**
- Auto-trigger only fires when a **new** email is submitted — it doesn't help for emails already queued but stuck/unprocessed
- If the auto-trigger subprocess silently fails (e.g. file permissions, `OSError`), the admin needs a fallback to manually re-trigger
- On `/manage/emails`, admins monitor **all** queued emails — they need a way to kick processing for older stuck emails, not just the one they just submitted
- On `finished.html`, the button is styled as secondary ("Re-trigger Processing", `btn-default`) with helper text — clearly communicating it's a fallback, not the primary action
- The `fcntl` file lock makes all of this completely safe — clicking the button while auto-trigger (or cron) is already running does nothing harmful

**Why subprocess instead of direct import/execfile:**
- `process_messages()` and `send_email_requests()` are explicitly **not thread-safe** (docstrings warn: "callers should ensure not called in more than one thread simultaneously")
- `send_email_requests()` sleeps 1.5 seconds between sends — a batch of 200 emails would block the HTTP response for 5+ minutes
- Subprocess is fully detached from the web process, so a crash in email processing never affects the admin's session
- Reuses `dbmail_cron.py`'s battle-tested `fcntl.lockf(LOCK_EX | LOCK_NB)` locking with zero new concurrency code

### Key changes

| File | Change |
|------|--------|
| `commmodule.py` | Auto-trigger `dbmail_cron.py` subprocess after `newmsg_request.save()` in `commfinal()` |
| `views.py` | New `process_emails` view — `@admin_required`, POST-only, returns JSON |
| `urls.py` | URL route at `/manage/process_emails` |
| `emails.html` | "Process Emails Now" button + AJAX handler on email monitoring dashboard |
| `finished.html` | Auto-trigger success banner + "Re-trigger Processing" fallback button + AJAX handler on commpanel finished page |
| `commpanel.py` | 13 new tests |

### Edge cases handled

| Edge case | How handled |
|-----------|-------------|
| Cron job already running | `fcntl` lock makes subprocess exit immediately (code 0) |
| Rapid button clicks | Button disabled during AJAX call, re-enabled on completion |
| Auto-trigger fails (e.g. `OSError`) | Wrapped in `try/except` — email is still saved, cron picks it up later |
| Non-admin access | `@admin_required` returns 302 (unauthenticated) or 403 (student/teacher) |
| Non-POST requests | Explicitly rejected with 400 (GET, PUT, DELETE) |
| Large email batches | Subprocess is fully detached, never blocks HTTP response |

## Related Issue

Closes #2920

## Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Documentation update
- [ ] Refactor / cleanup
- [ ] CI/CD or build change

## Testing

- [x] Existing tests passed
- [x] New tests added — 13 tests covering:
  - POST success with JSON response and correct `Content-Type`
  - HTTP method rejection (GET, PUT, DELETE → 400)
  - Authentication: anonymous → 302 redirect to login
  - Authorization: student → 403, teacher → 403
  - Subprocess mocking: verifies `sys.executable` + correct `dbmail_cron.py` path
  - Subprocess failure: `OSError` → 500 with JSON error body
  - Template integration: button HTML + AJAX URL present on `/manage/emails`
  - Auto-trigger on commpanel submit: full flow through `commpanel_old` → `commfinal`, verifies `Popen` called
  - Resilience: auto-trigger `OSError` does not break email submission, `MessageRequest` still saved
- [x] Manually verified 

### Email monitoring dashboard — "Process Emails Now" button

screenshot-emails-dashboard-button

### Email monitoring dashboard — success message after clicking

screenshot-emails-dashboard-success

### Email monitoring dashboard — with queued email visible

screenshot-emails-dashboard-with-email

### Commpanel finished page — auto-trigger banner + "Re-trigger Processing" fallback button

screenshot-commpanel-finished-button

### Commpanel finished page — success message after clicking "Re-trigger Processing"

screenshot-commpanel-finished-success

## AI Disclosure

This PR was developed with assistance from AI.

## Checklist

- [x] Self-reviewed the code
- [ ] Updated documentation (if needed)
- [x] No new warnings or errors introduced

### CODE REVIEW

File: esp/esp/program/views.py [hunk 1]
```diff
@@ -795,6 +798,60 @@
...
+            [sys.executable, dbmail_cron_path],
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.DEVNULL,
+            close_fds=True,
+            start_new_session=True,
+        )
+    except Exception as e:
+        logger.exception('Failed to launch dbmail_cron.py')
+        return HttpResponse(
+            json.dumps({'status': 'error', 'message': str(e)}),
```
[the remark this answers is not in the document]
Comment (abhinavjha0239) [reply esp/esp/program/views.py]: resolved

**Comment by abhinavjha0239** [comment]
@willgearty 

Rapid button clicks can spawn many short-lived Python processes (each loads Django before hitting the fcntl lock and exiting). Currently the button disables during the AJAX call, but re-enables on completion. Would a client-side cooldown (e.g. keep button disabled for a few seconds after response) be sufficient, or would you prefer server-side throttling too?

**Comment by willgearty** [comment]
I think client side is fine

**Comment by abhinavjha0239** [comment]
> I think client side is fine

Done ✅.
